src/main/java/hu/oe/nik/szfmv/automatedcar/model/WorldObject.java

```diff
@@ -140,64 +161,85 @@ public void setRotation(double rotation) {
      */
     public void setPosition(Position position) {
         this.position = position;
     }
 
     public String getImageFileName() {
-        return imageFileName;
+        return this.imageFileName;
     }
 
     public void setImageFileName(String imageFileName) {
         this.imageFileName = imageFileName;
     }
 
+    @Override
     public BufferedImage getImage() {
         return this.image;
     }
 
     @Override
-    public int getReferenceX()
-    {
+    public int getReferenceX() {
         return this.referencePosition.getX();
     }
 
     @Override
-    public int getReferenceY()
-    {
+    public int getReferenceY() {
         return this.referencePosition.getY();
     }
 
-    public void setReferencePosition(Position referencePosition)
-    {
+    public void setReferencePosition(Position referencePosition) {
         this.referencePosition = referencePosition;
     }
-    
-    @Deprecated
+
     public void initImage() {
         try {
-            image = ImageIO.read(new File(ClassLoader.getSystemResource(imageFileName).getFile()));
-            this.width = image.getWidth();
-            this.height = image.getHeight();
-        } catch (IOException e) {
-            LOGGER.error(e.getMessage());
+            this.image = ModelCommonUtil.loadObjectImage(this.getImageFileName());
+            this.width = this.image.getWidth();
+            this.height = this.image.getHeight();
+        } catch (Exception e) {
+            LOGGER.error(MessageFormat.format(Consts.ERROR_COULDNT_LOAD_IMG_FILE, this.getImageFileName()));
         }
     }
 
     /**
      * JAXB unmarshaller event listenerje.
      * Az objektum felépítése után hívódik meg, a kép betöltéséért felel.
      *
-     * @param u unmarshaller
+     * @param u      unmarshaller
      * @param parent JAXBElement
      */
     public void afterUnmarshal(Unmarshaller u, Object parent) {
-        try {
-            image = ModelCommonUtil.loadObjectImage(getImageFileName());
-        } catch (Exception e) {
-            LOGGER.error(MessageFormat.format(Consts.ERROR_COULDNT_LOAD_IMG_FILE, getImageFileName()));
-        }
+        this.initImage();
+        this.initShape();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Shape getPolygon() {
+        return this.getShapeTransfrom().createTransformedShape(this.polygon);
+    }
+
+    private AffineTransform getShapeTransfrom() {
+        double theta = Math.toRadians(this.getRotation());
+        AffineTransform shapeTransform = new AffineTransform();
+        shapeTransform.rotate(theta);
+        shapeTransform.translate(this.getPosX() + this.getReferenceX(), this.getPosX() + this.getReferenceY());
+
+        return shapeTransform;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void initShape() {
+        int x = 0 - (this.width / 2);
+        int y = 0 - (this.height / 2);
+        this.polygon = new Rectangle(x, y, this.width, this.height);
```

Comment:
ParkingSpace 39. sora miatt (meg amúgy is bármikor előfordulhat valami hiba, legyünk fail-safe-k) tényleg jó lenne, ha try-catchben lenne. A catch-ben meg logoljuk ki a stacktracet is, így: 

LOGGER.error("Nem sikerült a poligon beállítás a következő world objectnél: " + this.getImageFileName, e);`

Azért fontos, mert unmarshall után csináljuk, a JAXB meg elnyeli az exceptiont, aztán majd pislogunk miért nem mennek a dolgok.